### PR TITLE
Fixed issues with requests & added async request.

### PR DESCRIPTION
**Pull request by monocliffe**
[TPOZ-1735](https://wpengine.atlassian.net/browse/TPOZ-1735)

Fixed issues with requests & changed to async request. Also updated ComponentDidMount to useEffect. Switched from using axios library for requests to standard javascript requests.

### CODE REVIEW

File: src/Posts.js [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
+        fetchPostsData(props.id);
+    }, [props.id]);
+
+    return (
+        <ul>
+            { posts.map(post =>
+                <div key={post.id}>
+                    <div>
+                        <h3>{post.title}</h3>
+                        <div dangerouslySetInnerHTML={{__html: post.rich_body}}/>
```
Comment (JamesDominy) [inline src/Posts.js:27]: dangerously? Because it's not sanitised?

File: src/Posts.js [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
 
-    componentDidMount() {
-        axios.get('http://0.0.0.0:3000/posts.json')
-            .then(res => {
-                const posts = res.data;
-                this.setState({ posts });
-            })
+    async function fetchPostsData() {
+        const response = await fetch(
+            `http://0.0.0.0:3000/posts.json`
```
Comment (JamesDominy) [inline src/Posts.js:12]: `0.0.0.0` is a broadcast address? I assume this is meant to be `localhost`? Either way, you should generally avoid hard coded URLs in code and inject them from config, or dynamically construct them on the client side, e.g. by using `window.location` and its sub-propoerties

File: src/Posts.js (hunk 1 above)
Comment (monocliffe) [reply src/Posts.js:27]: Correct.

File: src/Posts.js (hunk 2 above)
Comment (monocliffe) [reply src/Posts.js:12]: Being handled here: [TPOZ-1737](https://wpengine.atlassian.net/browse/TPOZ-1737)

File: src/Posts.js (hunk 1 above)
Comment (p-lu) [reply src/Posts.js:27]: Yeah, I would look into sanitising it, or maybe breaking it down some other way to be injected

Comment (monocliffe) [reply src/Posts.js:27]: Being handled here: [TPOZ-1736](https://wpengine.atlassian.net/browse/TPOZ-1736)

**Review by JamesDominy — APPROVED** [review]

**Pull request merged**